vars/piperPipelineStageInit.groovy

```diff
@@ -172,6 +176,11 @@ void call(Map parameters = [:]) {
             .withMandatoryProperty('buildTool')
             .use()
 
+        if (config.transportRequestReqIDFromGit) {
```

Comment:
Good point

Reply:
To my understanding, we need to set` (config.transportRequestReqIDFromGit) `  during the initialization of piper i.e in `piperPipelineStageInit` otherwise we might miss this setting because our custom config and default config are mixed in during this step. What would you think?

Reply:
I dont understand your point. `config.transportRequestReqIDFromGit ` as well as  `config.transportRequestUploadCTS` are defined by the configuration. why should `config.transportRequestReqIDFromGit` get lost before `piperPipelineStageRelease` ? Well, provided that

```
stages:
  Release:
    transportRequestReqIDFromGit : true
    transportRequestUploadCTS: true
```

Reply:
I'd like to add that `transportRequestReqIDFromGit` is a step based on the initialization of piper. The Init should be done before!

Furthermore, it edits its result into the commonPipelineEnvironment and expects to stash the CPE for transportRequestUploadCTS. 

The Init stage, which creates the CPE and defines stashes, should not intervene.

Reply:
Actually, I dont know how/where the config settings

transportRequestReqIDFromGit : true 
transportRequestUploadCTS: true

are set-up! So, where are these settings done? And what is the difference between assigning it to the Release stage instead of the Init stage?

Reply:
These values come from the UI to the api service, which eventually marshalls it and send it to the pipeline, in pipeline we are writing these values in to our config file.

Setting it in Release stage would definately needs changes in the marshaller in the api service.

Reply:
That means, that the effort would be emense!

So instead, I would move the call to the end of the init stage procedure, right after `pipelineStashFilesBeforeBuild`, atleast after `piperInitRunStageConfiguration`